doc/macsec/MACsec_hld.md

```diff
@@ -0,0 +1,613 @@
+SAK           = 32HEXDIG / 64HEXDIG                 ; Secure Association Key.
+                                                    ; Default 128 bit,
+                                                    ; but if XPN enable, 258 bit
+AUTN_KEY      = 32HEXDIG                            ; The hash subkey in AES-GCM
+                                                    ; It's derived from SAK
+INIT_PN       = DIGITS                              ; 1 to 2^32-1, the initialized next packet number
+SALT          = 24HEXDIG                            ; 96-bit parameter provided to the Current
+                                                    ; Cipher Suite for subsequent protection
+                                                    ; and validation operations.
+                                                    ; Only available if XPN enable
+```
+
+#### 3.2.5 MACsec Ingress SA Table
+
+``` rfc5234
+"MACSEC_INGRESS_SA"|{{port_name}}|{{SCI}}|{{AN}}
+    "ACTIVE":{{TRUE|FALSE}}
+    "SAK":{{SAK}}
+    "AUTH_KEY":{{HASH_SUBKEY}}
+    "LOWEST_ACCEPTABLE_PN":{{PN}}
+    "SALT":{{SALT}}
+
+; Defines schema for MACsec Ingress SA table attributes
+key                   = MACSEC_INGRESS_SA:port_name:sci:an  ; MACsec SA identifier
+; field               = value
+ACTIVE                = "true" / "false"                    ; Whether this SA is available, Default False.
+SAK                   = 32HEXDIG / 64HEXDIG                 ; Secure Association Key.
+                                                            ; Default 128 bit,
+                                                            ; but if XPN enable, 258 bit
+AUTN_KEY              = 32HEXDIG                            ; The hash subkey in AES-GCM
+                                                            ; It's derived from SAK
+LOWEST_ACCEPTABLE_PN  = DIGITS                              ; 1 to 2^32-1, the lowest acceptable packet number
+SALT                  = 24HEXDIG                            ; 96-bit parameter provided to the Current
+                                                            ; Cipher Suite for subsequent protection
+                                                            ; and validation operations.
+                                                            ; Only available if XPN enable
+```
+
+### 3.3 State DB
+
+#### 3.3.1 MACsec Port Table
+
+``` rfc5234
+"MACSEC_PORT"|{{port_name}}
+    "state":{{OK}}
+
+; Defines schema for MACsec Port table attributes
+key          = MACSEC_PORT:port_name  ; Port name
+; field      = value
+state        = "" / "ok"              ; The MACsec port has readied to configure
+```
+
+#### 3.3.2 MACsec Egress SC Table
+
+``` rfc5234
+"MACSEC_EGRESS_SC"|{{port_name}}|{{SCI}}
+    "state":{{OK}}
+
+; Defines schema for MACsec Egress SC table attributes
+key          = MACSEC_EGRESS_SC:port_name:sci     ; MACsec SC identifier
+; field      = value
+state        = "" / "ok"                          ; The MACsec SC has been created
```

Comment:
I don't think "" is a value for state. Either it is present with "ok" or the table itself is not present. Can you check and correct here and the below ones?

Reply:
Done, please review it.